### ISSUE TITLE
Fix S2583: FP with null-coalescing operator

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonar-dotnet#2369]
Copied from https://github.com/SonarSource/sonarlint-visualstudio/issues/866 created by @peska

### Description

Warning S2583 Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.

### Repro steps

Sample class to reproduce. Warning in line 5:

```
public class TestClass
{
	public void TestMethod(bool? input)
	{
		if (input ?? false)
			Console.WriteLine("input is true");
		else
			Console.WriteLine("input is false");
	}
}
```

### Expected behavior

No warning given. This is a false positive.

### Actual behavior

Warning S2583 Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.

### Known workarounds

Use:

`if (input.GetValueOrDefault())`

### Related information

* SonarLint for VS Version: 4.9.0.4314 https://github.com/SonarSource/sonarlint-visualstudio/releases/tag/v4.9-rc2
* Visual Studio Version Enterprise 2019 16.0.0
* SonarQube server version: ---
* Languages in your solution:

**Comment by andrei-epure-sonarsource** [comment]
Probably #2361 , #2164, #1837 and #2369 are all related to how we interpret the null-conditional and null-coalesce operators

**Comment by andrei-epure-sonarsource** [comment]
I removed this from 7.15 milestone as there should be a separate milestone dedicated to fixing advanced rules which use Symbolic Execution
